src/node/stage/approved.rs

```diff
@@ -68,6 +69,13 @@ pub struct Approved {
     // Flag indicating that our section members changed (a node joined or left) and we might need
     // to change our elders.
     members_changed: bool,
+    // Map for DKG generators
+    key_gen_map: BTreeMap<(BTreeSet<PublicId>, u64), KeyGen<FullId>>,
+    dkg_ticks: Option<usize>,
+    pending_dkg_msg: BTreeMap<(BTreeSet<PublicId>, u64), VecDeque<Bytes>>,
+    pending_accumulated_events: VecDeque<(AccumulatingEvent, Proof)>,
+    completed_dkg: BTreeSet<(BTreeSet<PublicId>, u64)>,
+    sibling_dkg_result_cache: BTreeMap<BTreeSet<PublicId>, DkgResultWrapper>,
```

Comment:
Quite a lot of stuff added here. To maintain better separation of concerns, I'd consider moving this (and also the new dkg-related methods) over to `SectionKeysProvider` (or a separate `Dkg` module perhaps if that makes more sense).

Reply:
I have updated BLS_DKG to carry out caching internally, so some shall be removed.
Also, I will squash some into one struct once all tests settled.